### PR TITLE
Display if a mod deleted a message when it's logged

### LINKED ISSUE #60: Check the audit log to see if a mod deleted a message.
It appears that the audit log is open to the bot api, so why not check the latest entry when a message has been deleted, and see if it was done by a user, or a mod? If a mod deleted it, they could be added somewhere in the message.

> <user pfp> Message Deleted by mod#1234

And then continue like normal.

### PR DESCRIPTION
**Pull request by Tanish2002**
Fixes #60
it just works plox don't question me :crying_cat_face:, also this is my first golang pr so expect it to be bad :crying_cat_face:

### CODE REVIEW

File: messagelogger.go [hunk 1]
```diff
@@ -32,7 +33,27 @@ func messageDelete(s *discordgo.Session, m *discordgo.MessageDelete) {
 			log.Println("Recovered from panic in messageDelete", r)
 		}
 	}()
-
+	var deleter string
+	// get audit log info
+	auditLog, err := s.GuildAuditLog(m.GuildID, "", "", discordgo.AuditLogActionMessageDelete, 1)
+	if err != nil {
+		log.Printf("Failed to check audit log: %s", err.Error())
```
Comment (mlvzk) [inline messagelogger.go:40]: return; after this, the else won't be needed then

File: messagelogger.go [hunk 2]
```diff
@@ -32,7 +33,27 @@ func messageDelete(s *discordgo.Session, m *discordgo.MessageDelete) {
...
+		return
+	}
+	// get audit log entries
+	for _, v := range auditLog.AuditLogEntries {
```
Comment (elkowar) [inline messagelogger.go]: As you are only requesting a single Entry from the log, using a loop here is somewhat confusing.
I'd propose something like this:
```go
	if len(auditLog.AuditLogEntries) == 0 {
		log.Printf("Received no audit-log entries")
	}
	lastEntry := auditLog.AuditLogEntries[0]
	if lastAuditIds[m.GuildID] == lastEntry.ID {
		deleter = "self"
	} else {
		lastAuditIds[m.GuildID] = lastEntry.ID

		for _, u := range auditLog.Users {
			if u.ID == lastEntry.UserID {
				deleter = u.Username + "#" + u.Discriminator
			}
		}
	}

```

**Comment by buffet** [comment]
LGTM!!

Elko takes responsibility.

**Comment by elkowar** [comment]
looks good to me, but responsibility is actually on buffet, as he has the longer beard.

**Comment by elkowar** [comment]
Thanks!

**Comment by eepykate** [comment]
i have no beard so you must share the responsibility.

**Comment by eepykate** [comment]
@buffet lgtm? lesbian gay trans marriage?

**Comment by buffet** [comment]
@elkowar IDK sounds like beardiscrimination to me.

**Pull request merged**